src/Command/DumpEncryptedFieldsMapCommand.php

```diff
@@ -63,24 +60,25 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $dumper = new Dumper();
 
         foreach ($this->documentManagers as $name => $documentManager) {
-            $generator          = new EncryptedFieldsMapGenerator($documentManager->getMetadataFactory());
-            $encryptedFieldsMap = $generator->getEncryptedFieldsMap();
+            $encryptedFieldsMap = [];
+            foreach ($documentManager->getMetadataFactory()->getAllMetadata() as $metadata) {
+                $database               =  $documentManager->getDocumentDatabase($metadata->getName());
+                $collectionInfoIterator = $database->listCollections(['filter' => ['name' => $metadata->getCollection()]]);
+
+                foreach ($collectionInfoIterator as $collectionInfo) {
+                    if ($collectionInfo['options']['encryptedFields'] ?? null) {
+                        $encryptedFieldsMap[$this->getDocumentNamespace($metadata, $database->getDatabaseName())] = $collectionInfo['options']['encryptedFields'];
+                    }
+                }
+            }
 
             if (empty($encryptedFieldsMap)) {
                 continue;
             }
 
-            $encryptedFieldsMap = array_combine(
-                // Convert class names in keys to their full namespaces
-                array_map(
-                    fn (string $fqcn): string => $this->getDocumentNamespace(
-                        $documentManager->getClassMetadata($fqcn),
-                        $documentManager->getConfiguration()->getDefaultDB(),
-                    ),
-                    array_keys($encryptedFieldsMap),
-                ),
-                array_values($encryptedFieldsMap),
-            );
+            foreach ($encryptedFieldsMap as $ns => $encryptedFields) {
+                $encryptedFieldsMap[$ns] = json_decode(PackedArray::fromPHP($encryptedFields['fields'])->toRelaxedExtendedJSON(), true);
```

Comment:
I presume this back and forth is to ensure that we've got a PHP representation of relaxed extended JSON that the dumper can then dump in the selected format?

Reply:
Tracked in [PHPORM-371](https://jira.mongodb.org/browse/PHPORM-371)

Reply:
I assume the last comment was referring to the previous comment thread about merging fields when inheritance is used. Thanks for confirming the encode/decode behaviour.